### PR TITLE
Delegation

### PR DESCRIPTION
**Pull request by andorp**

### CODE REVIEW

File: src/Types.hs [hunk 1]
```diff
@@ -285,12 +285,21 @@ type instance Proto IdeaLike = ()
 data IdeaVote = IdeaVote
     { _ideaVoteMeta  :: MetaInfo IdeaVote
     , _ideaVoteValue :: IdeaVoteValue
+    , _ideaVoteVoter :: AUID User
```
Comment (fisx) [inline src/Types.hs]: rename to `_ideaVoteDelegate`?  also document which of delegate and delegator is stored in meta info.

Comment (fisx) [inline src/Types.hs]: also comment on the fact that the delegation path can be lost over time.  (i don't see a problem with it now, but it should be made explicit.)

File: src/Types.hs [hunk 2]
```diff
@@ -740,6 +749,15 @@ data DelegationContext =
 
 instance SOP.Generic DelegationContext
 
+data DelegationContextFull =
```
Comment (fisx) [inline src/Types.hs]: rename `DelegationContext` to `DScope` in all its variants?

File: tests/DelegationSpec.hs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        spaces = DlgCtxIdeaSpace <$> unIdeaSpaces u
+        topics = DlgCtxTopicId   <$> unTopics     u
+        ideas  = DlgCtxIdeaId    <$> unIdeas      u
+
+-- * delegation program
+
+data DelegationDSL where
+    SetDelegation       :: AUID User -> DelegationContext -> AUID User      -> DelegationDSL
+    Vote                :: AUID User -> AUID Idea         -> IdeaVoteValue  -> DelegationDSL
+    CheckNoOfDelegatees :: AUID User -> DelegationContext -> Int -> DelegationDSL
```
Comment (fisx) [inline tests/DelegationSpec.hs]: rename to `CheckNumVotingPower`.  it's number of delegatees (aka delegators) plus oneself.

**Comment by andorp** [comment]
Closing in favour of #629

**Pull request closed without merging**